### PR TITLE
added boost to setup instructions and refactored visualizer_rllib

### LINKED ISSUE #159: Clarify boost setup for Ray for Mac
When installing Ray from Eugene's repo in Mac, I ran into the issue of needing to install boost before I could run `python setup.py develop` (I got the `fatal error: 'boost/algorithm/string/predicate.hpp' file not found`).

Boost is not mentioned in Eugene's repo [Ray installation deps ](https://github.com/eugenevinitsky/ray/blob/master/doc/source/installation.rst) or in the official [Ray installation deps](https://ray.readthedocs.io/en/latest/installation.html)

`brew install boost` did not work as brew does not have a formula for Boost 1.6.5, and any other version led to some missing files.

For me to install Boost I had to download Boost 1.6.5, unzip it, and manually place the "boost" folder into my `/usr/local/include/` folder. I got the version number from [this setup script for boost in the ray repo](https://github.com/ray-project/ray/blob/master/thirdparty/scripts/build_boost.sh)

So perhaps we could clarify the above steps, and write a script to automate the process? Also when I tried to run the above setup script for boost it didn't install boost correctly for me, and I'm not sure how it's actually used and for what.

### PR DESCRIPTION
**Pull request by kev-chien**
closes #159 and addresses Eugene's comment in #188

### CODE REVIEW

**Comment by eugenevinitsky** [comment]
Think this is failing for reasons not to do with you, once we merge some higher up PRs this should pass.

File: docs/source/flow_setup.rst [hunk 1]
```diff
@@ -287,6 +287,9 @@ required libraries as specified at
 <http://ray.readthedocs.io/en/latest/installation.html> and
 then follow the setup instructions.
 
+Of note is that boost is a dependency not mentioned in the additional libraries.
+Boost 1.6.5 is required, you can download the zip `here <https://dl.bintray.com/boostorg/release/1.65.0/source/boost_1_65_0.zip>`_.
```
Comment (eugenevinitsky) [inline docs/source/flow_setup.rst]: @kev-chien I think boost can be installed via apt-get or brew; you don't need to set it up from source?

File: flow/visualize/visualizer_rllib.py [hunk 2]
```diff
@@ -91,8 +91,7 @@
     register_env(env_name, create_env)
 
     # Determine agent and checkpoint
-    config_run = config['env_config']['run'] if 'run' in config['env_config'] \
-        else None
+    config_run = config['env_config'].get('run', None)
```
Comment (eugenevinitsky) [inline flow/visualize/visualizer_rllib.py]: Nice

**Comment by AboudyKreidieh** [comment]
this has become stale, so closing. please feel free to reopen if needed.

**Pull request closed without merging**